### DISCUSSION TITLE
Specifying Configuration File

### DISCUSSION THREAD

**Discussion by rdlnz** [wifinigel/wiperf#66] [Ideas]
Hi Nigel,

Would there be any great challenge to allowing the configuration file to be passed as an argument to wiperf_run.py?

I was wondering about running tests on both the wireless and wired interfaces and thought running wiperf twice with different configuration files could potentially allow this.

I can envisage two issues:
1. Routing. I'm guessing the testing for one interface would need to use specific static routes for the required destination/s while the other had default OR wiperf would need to move the default route at test time. I've noticed that wiperf does sometimes add a default when certain tests fail, so maybe this isn't quite as hard as I imagine.

2. Reporting. At the moment I'm guessing the hostname is used to distinguish data from different probes. If it was possible to set that in the configuration file instead then the two instances could differentiate themselves e.g. hostname-eth0 and hostname-wlan0.

I also briefly imagined multiple config files might allow for 5 GHz and 2.4 GHz testing, but as far as I understand it that is controlled by /etc/network/interfaces, so that's probably even more work.

Many thanks for all your work. Much appreciated.
Regards,
Ray

**Comment by wifinigel** [comment]
Sorry for the latest response on this Ray.

I think the main challenge is the routing aspect. It can be a real struggle to get the routing to work correctly in some connection scenarios for a single connection, so I think that trying to cater for 2 active interfaces would be a big ask. 

I must admit that I really like the idea you have suggested, but it would take quite a bit of head-scratching to get a real-world implementation going.  I am in process of writing version 3 of the probe which supports IPv6, which make the routing consideration even more of a challenge.

I'm not saying no to this suggestion, but it would probably be a longer term goal for the project if I'm honest.

I definitely won't forget about this one, but it will be some way down the line.

Thanks again for the suggestion, I really appreciate it.

All the best.

Nigel.
